src/main/java/ch/njol/skript/expressions/ExprFallenDistance.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Fall Distance")
+@Description({"The distance an entity has fallen for"})
+@Examples({"set all entities' fall distance to 10",
+	"on damage:",
+	"\tsend \"%victim's fall distance%\" to victim"})
+@Since("INSERT VERSION")
+public class ExprFallenDistance extends SimplePropertyExpression<Entity, Number> {
+	
+	static {
+		register(ExprFallenDistance.class, Number.class, "[the] fall[en] (distance|height)", "entities");
+	}
+	
+	@Nullable
+	@Override
+	public Number convert(Entity entity) {
+		return entity.getFallDistance();
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.RESET || mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.DELETE) ? null : CollectionUtils.array(Number.class);
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta != null) {
+			Entity[] entities = getExpr().getArray(e);
+			if (entities.length < 1)
+				return;
+			Float number = ((Number) delta[0]).floatValue();
+			for (Entity entity : entities) {
+				
+				Float fallDistance = entity.getFallDistance();
+				
+				switch (mode) {
+					case ADD:
+						entity.setFallDistance(fallDistance + number);
+						break;
+					case SET:
+						entity.setFallDistance(number);
+						break;
+					case REMOVE:
+						entity.setFallDistance(fallDistance - number);
+						break;
```

Comment:
What happens if ``x`` is passed onto ``entity.setFallDistance(x)`` where ``x`` is negative? E.g. when the user gives a negative number, or they try and subtract too much from the entity's original ``fallDistance``?

Reply:
Not much, bukkit handles it as if it were 0

Reply:
Did you actually test this? Because sometimes Bukkit may be absolutely fine with it but then when it gets passed down the chain into NMS Minecraft starts throwing exceptions...

Reply:
Yeah I tested it